src/AutoMerge/guidelines.jl

```diff
@@ -91,8 +98,15 @@ function meets_compat_for_all_deps(working_directory::AbstractString, pkg, versi
     for version_range in keys(deps)
         if version in Pkg.Types.VersionRange(version_range)
             for name in keys(deps[version_range])
-                if !is_jll_name(name)
-                    @debug("Found a new (non-JLL) dependency: $(name)")
+                if _AUTOMERGE_REQUIRE_STDLIB_COMPAT
+                    debug_msg = "Found a new (non-JLL) dependency: $(name)"
+                    apply_compat_requirement = !is_jll_name(name)
+                else
+                    debug_msg = "Found a new (non-stdlib non-JLL) dependency: $(name)"
+                    apply_compat_requirement = !is_jll_name(name) && !is_julia_stdlib(name)
```

Comment:
I don't think this function exists anymore, we'd need to restore it

Reply:
Indeed. Restored in https://github.com/JuliaRegistries/RegistryCI.jl/pull/540/commits/46b5d7069710a05e0ea2fd385918db23ab7f48ae